core/src/main/java/org/infinispan/health/impl/ClusterHealthImpl.java

```diff
@@ -0,0 +1,53 @@
+package org.infinispan.health.impl;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.infinispan.health.ClusterHealth;
+import org.infinispan.health.HealthStatus;
+import org.infinispan.manager.EmbeddedCacheManager;
+
+public class ClusterHealthImpl implements ClusterHealth {
```

Comment:
Yes and you specify which cache manager in the CLI - e.g. `/subsystem/datagrid-infinispan/cache-container/clustered/health/HEALTH`. 

So in this context when I'm asking is my cluster healthy, I mean if all nodes in the cluster operate correctly and if cache is not in degraded mode. 

I believe the terminology is correct.